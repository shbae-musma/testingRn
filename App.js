/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react'
import { View, Text } from 'react-native'
import User from './src/User'

const App = () => {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>아래에 컴포넌트가 들어갑니다.</Text>
      <User name={'악당별'} id={'a9603'} />
    </View>
  );
};

export default App