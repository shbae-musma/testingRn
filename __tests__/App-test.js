/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
// import renderer from 'react-test-renderer';
// it('renders correctly', () => {
//   renderer.create(<App />);
// });

import { render } from '@testing-library/react-native'

let props
let component

function getComponent(props) {
  return <App {...props} />
}

describe('App test...', () => {
  props = {}
  component = getComponent(props)
  test('test 1', () => {
    const rendering = render(component)
    expect(rendering).toMatchSnapshot()
    expect(rendering).toBeTruthy()
  })
})