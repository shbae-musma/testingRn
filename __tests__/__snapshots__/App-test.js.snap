// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test... test 1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    아래에 컴포넌트가 들어갑니다.
  </Text>
  <View>
    <Text>
      메시지가 표시 됩니다. : 
      Who a u?
    </Text>
    <Text>
      아이디가 표시 됩니다 : 
      a9603
    </Text>
    <Text>
      이름이 표시 됩니다 : 
      악당별
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          버튼입니다.
        </Text>
      </View>
    </View>
  </View>
</View>
`;
